corebc-core/src/types/network.rs

```diff
@@ -1,36 +1,18 @@
 use super::{U128, U256, U512, U64};
-use serde::{Deserialize, Serialize, Serializer};
-use std::{
-    convert::{TryFrom, TryInto},
-    fmt,
-    time::Duration,
-};
-use strum::{AsRefStr, EnumCount, EnumIter, EnumString, EnumVariantNames};
-
-// compatibility re-export
-#[doc(hidden)]
-pub use num_enum::{TryFromPrimitive, TryFromPrimitiveError};
-#[doc(hidden)]
-pub type ParseNetworkError = TryFromPrimitiveError<Network>;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{convert::TryFrom, fmt, str::FromStr, time::Duration};
+use strum::{EnumCount, EnumIter, EnumVariantNames};
+
+#[derive(Debug)]
+pub struct ParseNetworkError {
+    pub number: u64,
+}
 
 // When adding a new network:
 //   1. add new variant to the Network enum;
 //   2. add extra information in the last `impl` block (explorer URLs, block time) when applicable;
-//   3. (optional) add aliases:
-//     - Strum (in kebab-case): `#[strum(to_string = "<main>", serialize = "<aliasX>", ...)]`
-//      `to_string = "<main>"` must be present and will be used in `Display`, `Serialize`
-//      and `FromStr`, while `serialize = "<aliasX>"` will be appended to `FromStr`.
-//      More info: <https://docs.rs/strum/latest/strum/additional_attributes/index.html#attributes-on-variants>
-//     - Serde (in snake_case): `#[serde(alias = "<aliasX>", ...)]`
-//      Aliases are appended to the `Deserialize` implementation.
-//      More info: <https://serde.rs/variant-attrs.html>
 //     - Add a test at the bottom of the file
 
-// We don't derive Serialize because it is manually implemented using AsRef<str> and it would
-// break a lot of things since Serialize is `kebab-case` vs Deserialize `snake_case`.
-// This means that the Network type is not "round-trippable", because the Serialize and Deserialize
-// implementations do not use the same case style.
-
```

Comment:
Did you find out why is this the case? It doesn't really make sense to me to make the network type not "round-trippable"

Reply:
not sure about that, but thats what I found about this 
https://github.com/gakonst/ethers-rs/issues/2038#issuecomment-1377773079
https://github.com/gakonst/ethers-rs/pull/2268#discussion_r1138931364

https://github.com/gakonst/ethers-rs/pull/2270/files
Seems like it was hot-fixed and tests for "round-trippable" were added